cpp/src/arrow/CMakeLists.txt

```diff
@@ -422,7 +422,9 @@ if(ARROW_COMPUTE)
        compute/kernels/scalar_nested.cc
        compute/kernels/scalar_random.cc
        compute/kernels/scalar_set_lookup.cc
-       compute/kernels/scalar_string.cc
+       compute/kernels/scalar_string_common.cc
```

Comment:
The current convention is to suffix file names with `_internal` when its contents are local common utilities. Based on this, I suggest `scalar_string_common.cc` --> `scalar_string_internal.cc`.

Reply:
Okay, done in 36e134b. Based on other `*_internal.h` files, I guess that will also satisfy the linter w.r.t. `RETURN_NOT_OK` macro usage in the corresponding header file?